include/RED4ext/Relocation-inl.hpp

```diff
@@ -27,50 +25,145 @@ uintptr_t RED4ext::UniversalRelocBase::Resolve(uint32_t aHash)
     }
     else
     {
-        using functionType = uintptr_t (*)(uint32_t);
-        static functionType resolveFunc = nullptr;
-
-        static std::once_flag flag;
-        std::call_once(flag,
-                       []()
-                       {
-                           constexpr auto dllName = "RED4ext.dll";
-                           constexpr auto functionName = "RED4ext_ResolveAddress";
-
-                           auto handle = GetModuleHandleA(dllName);
-                           if (!handle)
-                           {
-                               std::stringstream stream;
-                               stream << "Failed to get '" << dllName
-                                      << "' handle.\nProcess will now close.\n\nLast error: " << GetLastError();
-
-                               MessageBoxA(nullptr, stream.str().c_str(), "RED4ext.SDK", MB_ICONERROR | MB_OK);
-                               TerminateProcess(GetCurrentProcess(), 1);
-                           }
-
-                           resolveFunc = reinterpret_cast<functionType>(GetProcAddress(handle, functionName));
-                           if (resolveFunc == nullptr)
-                           {
-                               std::stringstream stream;
-                               stream << "Failed to get '" << functionName
-                                      << "' address.\nProcess will now close.\n\nLast error: " << GetLastError();
-
-                               MessageBoxA(nullptr, stream.str().c_str(), "RED4ext.SDK", MB_ICONERROR | MB_OK);
-                               TerminateProcess(GetCurrentProcess(), 1);
-                           }
-                       });
+        auto resolveFunc = GetAddressResolverFunction();
 
         auto address = resolveFunc(aHash);
         if (address == 0)
         {
-            std::stringstream stream;
-            stream << "Failed to resolve address for hash " << std::hex << std::showbase << aHash
-                   << ".\nProcess will now close.";
+            std::wostringstream stream;
+            stream << L"Failed to find the address for the hash (" << std::dec << aHash << ") provided by the plugin.\n"
+                   << L"This issue is likely caused by the mod using an incorrect or outdated hash.";
 
-            MessageBoxA(nullptr, stream.str().c_str(), "RED4ext.SDK", MB_ICONERROR | MB_OK);
-            TerminateProcess(GetCurrentProcess(), 1);
+            ShowErrorAndTerminateProcess(stream.str(), GetLastError());
         }
 
         return address;
     }
 }
+
+RED4EXT_INLINE HMODULE RED4ext::UniversalRelocBase::GetCurrentModuleHandle()
+{
+    HMODULE result;
+
+    if (!GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS | GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
```

Comment:
unsure if you define UNICODE quickly but beware this may be A or W depedning on that

Reply:
you use TCHAR so i suppose it may be fine but considering you use W funcs explicitly everywhere, would maybe consider making this explicit just in case too